File: cpp/p29.cpp
#include "euler.h"

using namespace std;

int main() {
  map<double, int> terms;
	
  for( int a = 2; a <= 100; a++ )
    for( int b = 2; b <= 100; b++ )
      terms[ pow( a, b ) ] = 1;
	
  cout << terms.size() << endl;
}

File: clojure/p29.clj
(ns p29
  (:use [euler :only (expt timed-test)]))

(timed-test
 9183
 (let [expts (for [a (range 2 101)
                   b (range 2 101)]
               (expt a b))]
   (count (distinct expts))))
